tests/e2e/mnist_raycluster_sdk_test.go

```diff
@@ -0,0 +1,242 @@
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+						{
+							Name: "workdir",
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name: "test",
+							// FIXME: switch to base Python image once the dependency on OpenShift CLI is removed
+							// See https://github.com/project-codeflare/codeflare-sdk/pull/146
+							Image: "quay.io/opendatahub/notebooks:jupyter-minimal-ubi8-python-3.8-4c8f26e",
+							Env: []corev1.EnvVar{
+								{Name: "PYTHONUSERBASE", Value: "/workdir"},
+								{Name: "RAY_IMAGE", Value: GetRayImage()},
+							},
+							Command: []string{
+								"/bin/sh", "-c",
+								"while [ ! -f /codeflare-sdk/pyproject.toml ]; do sleep 1; done; " +
+								"cp /test/* . && chmod +x install-codeflare-sdk.sh && ./install-codeflare-sdk.sh && python mnist_raycluster_sdk.py " + namespace.Name,
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "test",
+									MountPath: "/test",
+								},
+								{
+									Name:      "codeflare-sdk",
+									MountPath: "/codeflare-sdk",
+								},
+								{
+									Name:      "workdir",
+									MountPath: "/workdir",
+								},
+							},
+							WorkingDir: "/workdir",
+							SecurityContext: &corev1.SecurityContext{
+								AllowPrivilegeEscalation: Ptr(false),
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: "RuntimeDefault",
+								},
+								Capabilities: &corev1.Capabilities{
+									Drop: []corev1.Capability{"ALL"},
+								},
+								RunAsNonRoot: Ptr(true),
+							},
+						},
+					},
+					RestartPolicy:      corev1.RestartPolicyNever,
+					ServiceAccountName: sa.Name,
+				},
+			},
+		},
+	}
+	if GetClusterType(test) == KindCluster {
+		// Take first KinD node and redirect pod hostname requests there
+		node := GetNodes(test)[0]
+		hostname := GetClusterHostname(test)
+		IP := GetNodeInternalIP(test, node)
+
+		test.T().Logf("Setting KinD cluster hostname '%s' to node IP '%s' for SDK pod", hostname, IP)
+		job.Spec.Template.Spec.HostAliases = []corev1.HostAlias{
+			{
+				IP:        IP,
+				Hostnames: []string{hostname},
+			},
+		}
+
+		// Propagate hostname into Python code as env variable
+		hostnameEnvVar := corev1.EnvVar{Name: "CLUSTER_HOSTNAME", Value: hostname}
+		job.Spec.Template.Spec.Containers[0].Env = append(job.Spec.Template.Spec.Containers[0].Env, hostnameEnvVar)
+	}
+	job, err := test.Client().Core().BatchV1().Jobs(namespace.Name).Create(test.Ctx(), job, metav1.CreateOptions{})
+	test.Expect(err).NotTo(HaveOccurred())
+	test.T().Logf("Created Job %s/%s successfully", job.Namespace, job.Name)
+
+	go func() {
+		// Checking if pod is found and running
+		podName := ""
+		foundPod := false
+		for !foundPod {
+			pods, _ := test.Client().Core().CoreV1().Pods(namespace.Name).List(test.Ctx(), metav1.ListOptions{
```

Comment:
I'd rather avoid that go routine with a for loop that's hard to read. As it's for waiting for the Pod to be available, I'd suggest to either:
- use a "raw" Pod instead of a Job to run the SDK
- or use `Eventually` to do the polling until the Pod can be found.